torch/csrc/jit/passes/quantization.cpp

```diff
@@ -1277,26 +1328,49 @@ void insertQuantDeQuantCall(
   } else {
     quantize_func = "quantize_per_tensor";
   }
-  Node* quant = g->create(at::Symbol::aten(quantize_func), inputs);
-  quant->output()->setDebugName(v->debugName() + ".quant");
-  g->insertNode(quant);
+  Node *quant, *choose_qparams;
+  if (is_dynamic && !isWeightOfConvOrLinear(v)) {
+    std::string choose_qparams_func = "_choose_qparams_per_tensor";
+    auto reduce_range = g->insertConstant(false);
+    choose_qparams = g->create(at::Symbol::aten(choose_qparams_func), v, 2);
+    choose_qparams->addInput(reduce_range);
+
+    choose_qparams->output(0)->setDebugName(v->debugName() + ".scale_");
+    choose_qparams->output(0)->setType(FloatType::get());
+    choose_qparams->output(1)->setDebugName(v->debugName() + ".zero_point");
+    choose_qparams->output(1)->setType(IntType::get());
 
+    g->insertNode(choose_qparams);
+
+    std::vector<Value*> quant_inputs = {v};
+    for (auto& out : choose_qparams->outputs()) {
+      quant_inputs.push_back(out);
+    }
+    // Last argument is dtype.
+    quant_inputs.push_back(inputs.back());
+    quant = g->create(at::Symbol::aten(quantize_func), quant_inputs);
+    g->insertNode(quant);
+  } else {
+    quant = g->create(at::Symbol::aten(quantize_func), inputs);
```

Comment:
could you move the construction of `inputs` to this branch?

Reply:
I'm using the `dtype` information from the inputs in the `quant_inputs`

Reply:
you can still move that inside this branch and create a getattr node for dtype for `quant_inputs`, this makes me think that we probably want to change `qparam_names` to be a map as well, but that can be a different PR